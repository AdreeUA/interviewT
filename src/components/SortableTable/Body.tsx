import React from 'react'
import * as _ from 'lodash'

interface Props {
  data: any[]
}

interface State {}

class Body extends React.Component<Props, State> {
  render() {
    const { data } = this.props
    return (
      <tbody>
        {!_.isEmpty(data) &&
          data.map(({ fullName, thumbnail, email, phone }) => {
            return (
              <tr key={email}>
                <td>
                  <div className="table__image-wrapper">
                    <img src={thumbnail} className="table__image" alt={fullName} />
                  </div>
                </td>
                <td className="table__name">{fullName}</td>
                <td>{email}</td>
                <td>{phone}</td>
              </tr>
            )
          })}
      </tbody>
    )
  }
}

export default Body
