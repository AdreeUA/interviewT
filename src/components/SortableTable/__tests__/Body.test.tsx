import * as React from 'react'
import { shallow } from 'enzyme'
import userStateMock from '../../../__mocks__/usersStateMock'
import Body from '../Body'

describe('Body', () => {
  it('render() whith initialState', () => {
    const wrapper = shallow(<Body data={[]} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('render() whith compteted state', () => {
    const wrapper = shallow(<Body data={userStateMock.filtersResults} />)
    expect(wrapper).toMatchSnapshot()
  })
})
