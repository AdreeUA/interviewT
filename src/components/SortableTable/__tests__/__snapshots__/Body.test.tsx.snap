// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body render() whith compteted state 1`] = `
<tbody>
  <tr
    key="ariane.scott@example.com"
  >
    <td>
      <div
        className="table__image-wrapper"
      >
        <img
          alt="mrs ariane scott"
          className="table__image"
          src="https://randomuser.me/api/portraits/thumb/women/5.jpg"
        />
      </div>
    </td>
    <td
      className="table__name"
    >
      mrs ariane scott
    </td>
    <td>
      ariane.scott@example.com
    </td>
    <td>
      563-392-7975
    </td>
  </tr>
</tbody>
`;

exports[`Body render() whith initialState 1`] = `<tbody />`;
